tests/components/victron_remote_monitoring/test_sensor.py

```diff
@@ -87,9 +28,39 @@ async def test_unique_ids(
         "energy_production_estimate_today",
         "consumption_highest_peak_time_today",
     ):
-        unique_id = f"{key}_{site_id}"
+        unique_id = f"{site_id}|{key}"
         entity_id = ent_reg.async_get_entity_id("sensor", DOMAIN, unique_id)
         assert entity_id is not None, key
         entity = ent_reg.async_get(entity_id)
         assert entity is not None, entity_id
         assert entity.unique_id == unique_id
+
+
+async def test_sensors_snapshot(
+    hass: HomeAssistant,
+    init_integration,
+    mock_config_entry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Snapshot all VRM sensor states & key attributes."""
+    ent_reg = er.async_get(hass)
+    site_id = mock_config_entry.data["site_id"]
+    collected: dict[str, dict] = {}
+    for entry in ent_reg.entities.values():
+        if entry.platform != DOMAIN:
+            continue
+        if not entry.unique_id.startswith(f"{site_id}|"):
+            continue
+        state_obj = hass.states.get(entry.entity_id)
+        assert state_obj is not None, entry.entity_id
+        attrs = state_obj.attributes
+        collected[entry.unique_id] = {
+            "entity_id": entry.entity_id,
+            "state": state_obj.state,
+            "device_class": attrs.get("device_class"),
+            "state_class": attrs.get("state_class"),
+            "unit_of_measurement": attrs.get("unit_of_measurement"),
+            "suggested_display_precision": attrs.get("suggested_display_precision"),
+        }
+    collected = dict(sorted(collected.items()))
+    assert collected == snapshot
```

Comment:
Use the snapshot platform helper instead

```py
await snapshot_platform(
            hass, entity_registry, snapshot, mock_cloud_config_entry.entry_id
        )
```
This also removes the need for a lot of other tests like testing the unique id